pkg/snmp/traps/server.go

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020 Datadog, Inc.
+
+package traps
+
+import (
+	"net"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/soniah/gosnmp"
+)
+
+// SnmpPacket is the type of packets yielded by server listeners.
+type SnmpPacket struct {
+	Content *gosnmp.SnmpPacket
+	Addr    *net.UDPAddr
+}
+
+// PacketsChannel is the type of channels of trap packets.
+type PacketsChannel = chan *SnmpPacket
+
+// TrapServer manages an SNMPv2 trap listener.
+type TrapServer struct {
+	Addr     string
+	config   *Config
+	listener *gosnmp.TrapListener
+	packets  PacketsChannel
+}
+
+var (
+	outputChannelSize = 100
+	server            *TrapServer
+	startError        error
+)
+
+// StartServer starts the global trap server.
+func StartServer() error {
+	s, err := NewTrapServer()
+	server = s
+	startError = err
+	return err
+}
+
+// StopServer stops the global trap server, if it is running.
+func StopServer() {
+	if server != nil {
+		server.Stop()
+		server = nil
+		startError = nil
+	}
+}
+
+// IsRunning returns whether the trap server is currently running.
+func IsRunning() bool {
+	return server != nil
+}
+
+// GetPacketsChannel returns a channel containing all received trap packets.
+func GetPacketsChannel() PacketsChannel {
+	return server.packets
+}
+
+// NewTrapServer configures and returns a running SNMP traps server.
+func NewTrapServer() (*TrapServer, error) {
+	c, err := ReadConfig()
```

Comment:
```suggestion
	config, err := ReadConfig()
```

(minor) c => config, improves readability. There are multiple locations where single letter is used, which is concise, but makes the code less readable.

IMO, we should avoid single letter vars and acronyms in general.

Reply:
From what I've seen single-letter variables isn't uncommon in the Golang world, but I agree. :-) I'll change them when it's obvious we can avoid them.